cishouseholds/validate.py

```diff
@@ -148,3 +148,21 @@ def check_singular_match(
             failure_column_name, F.when(F.col(existing_failure_column) == 1, 1).otherwise(F.col(failure_column_name))
         )
     return df.drop("TOTAL")
+
+
+def check_lookup_table_joined_columns_unique(df, join_column_list, name_of_df=""):
```

Comment:
This could be simplified to carry out the groupBy once:
```
duplicate_key_rows = df.groupBy(*join_column_list).count().filter("count > 1").drop("count")
if duplicate_key_rows.count() > 0:
    ...
```